src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -50,6 +51,192 @@ static void Main()
                 );
         }
 
+        [Fact]
+        public void SpeakableInference_MethodTypeInference()
```

Comment:
>SpeakableInference_MethodTypeInference [](start = 20, length = 38)

Are there any interesting scenarios worth testing with more than one input to the inference machinery? Applicable to array and lambda inference as well.

Reply:
Including a variations when inputs are different

---
In reply to: [243102042](https://github.com/dotnet/roslyn/pull/31813#discussion_r243102042) [](ancestors = 243102042)

Reply:
The variations that seem interesting: a T and tight-T, or a T and a U, in various situations (by-value, ref, or out, to trigger co/in/contra-variant rules) are covered pretty well already.

---
In reply to: [243102482](https://github.com/dotnet/roslyn/pull/31813#discussion_r243102482) [](ancestors = 243102482,243102042)